deploy_script.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+# Function to prompt user for yes/no input
+prompt_yes_no() {
+    while true; do
+        read -p "$1 (y/n): " yn
+        case $yn in
+            [Yy]* ) return 0;;
+            [Nn]* ) return 1;;
+            * ) echo "Please answer yes or no.";;
+        esac
+    done
+}
+
+echo "Welcome to the deployment script for your application!"
+echo "This script will help you set up your Ubuntu VPS with Nginx, Gunicorn, Certbot, and Supervisor."
+
+# Update system
+if prompt_yes_no "Do you want to update your system?"; then
+    echo "Updating system..."
+    sudo apt update && sudo apt upgrade -y
+fi
+
+# Install dependencies
+if prompt_yes_no "Do you want to install Python, Nginx, and Certbot?"; then
+    echo "Installing dependencies..."
+    sudo apt install python3 python3-pip python3-venv nginx certbot python3-certbot-nginx -y
+fi
+
+# Install Supervisor
+if prompt_yes_no "Do you want to install Supervisor?"; then
+    echo "Installing Supervisor..."
+    sudo apt install supervisor -y
+fi
+
+# Set up virtual environment and install requirements
+if prompt_yes_no "Do you want to set up a virtual environment and install requirements?"; then
+    echo "Setting up virtual environment..."
+    python3 -m venv venv
+    source venv/bin/activate
+    pip install -r requirements.txt
+    pip install gunicorn uvicorn
+fi
+
+# Configure Nginx
```

Comment:
@Aiderbot Let's use the domain name variable in the Nginx / certbot filenames rather than using 'mathweb'.

Reply:
@Aiderbot Can we make it so that it checks if the dependencies are installed before it installs each thing?